src/main/java/com/spotify/docker/client/DefaultDockerClient.java

```diff
@@ -1363,6 +1367,7 @@ public String build(final Path directory, final String name, final String docker
 
     WebTarget resource = noTimeoutResource().path("build");
 
+    final RegistryAuth registryAuth = registryAuthSupplier.authFor(name);
```

Comment:
Are we missing a test? Also, from my understanding, RegistryAuth only represents creds for one registry, not multiple, so I'm not sure how that would work on master right now